src/EventListener/InsertTagsListener.php

```
@@ -0,0 +1,106 @@
+    ];
+
+    /**
+     * Constructor.
+     *
+     * @param ContaoFrameworkInterface $framework
+     */
+    public function __construct(ContaoFrameworkInterface $framework)
+    {
+        $this->framework = $framework;
+    }
+
+    /**
+     * Replaces tags insert tags.
+     *
+     * @param string $tag
+     *
+     * @return string|false
+     */
+    public function onReplaceInsertTags($tag)
+    {
+        $elements = explode('::', $tag);
+        $key      = strtolower($elements[0]);
+
+        if (in_array($key, $this->supportedTags, true)) {
+            return $this->replaceInsertTag($key, $elements);
+        }
+
+        return false;
+    }
+
+
+    /**
+     * Replaces a tags-related insert tag.
+     *
+     * @param string $insertTag
+     * @param array $elements
+     *
+     * @return string
+     */
+    private function replaceInsertTag($insertTag, array $elements)
+    {
+        $this->framework->initialize();
+
+        $criteria['source'] = $elements[1];
+        $idOrAlias          = $elements[2];
+
+        /** @var TagModel $adapter */
+        $adapter = $this->framework->getAdapter(TagModel::class);
+
+        if (is_numeric($idOrAlias)) {
+            $criteria = [
+                'values' => [$idOrAlias]
+            ];
+        } else {
+            $criteria = [
+                'aliases' => [$idOrAlias]
+            ];
+        }
+
+        if (null === ($tags = $adapter->findByCriteria($criteria))) {
+            return '';
+        }
+
+        return $this->generateReplacement($tags, $insertTag);
+    }
+
+    /**
+     * Generates the replacement string.
+     *
+     * @param Collection $tags
+     * @param string $insertTag
+     *
+     * @return string
+     */
+    private function generateReplacement(Collection $tags, $insertTag)
+    {
+        switch ($insertTag) {
+            case 'tags_title':
+                return $tags->first()->name;
+                break;
+        }
+
+        return '';
```

I think we should throw an exception here and not return an empty string.